File: back/src/main/java/org/example/back/db/entity/Record.java
package org.example.back.db.entity;

import java.time.Duration;
import java.time.LocalDateTime;

import jakarta.persistence.*;
import lombok.*;
import org.example.back.common.BaseEntity;

import org.example.back.db.enums.GameMode;
import org.example.back.record.dto.RecordDto;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Builder
@Table(name = "record")
public class Record extends BaseEntity {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "id", nullable = false)
	private Long id;

	@ManyToOne(fetch = FetchType.LAZY, optional = false)
	@JoinColumn(name = "member_id", nullable = false)
	private Member member;

	@Lob
	@Enumerated(EnumType.STRING)
	@Column(name = "game_mode")
	private GameMode gameMode;

	@Column(name = "ranking")
	private Integer ranking;

	@Column(name = "distance")
	private Float distance;

	@Column(name = "run_start_time")
	private LocalDateTime runStartTime;

	@Column(name = "run_end_time")
	private LocalDateTime runEndTime;

	@Column(name = "pace")
	private Integer pace;

	@Column(name = "calorie")
	private Integer calorie;

	@Column(name="duration")
	private Integer duration;

	public Long getDuration() {
		if (runStartTime != null && runEndTime != null) {
			return Duration.between(runStartTime, runEndTime).toMillis();
		} else {
			return null;
		}
	}
	public RecordDto toRecordDto() {
		return RecordDto.builder()
				.id(getId())
				.duration(getDuration())
				.gameMode(getGameMode())
				.ranking(getRanking())
				.distance(getDistance())
				.build();
	}

}

File: front_android/lib/src/model/record.dart
class Record {
  String? courseImgUrl;
  int? id;
  String? runStartTime;
  String? runEndTime;
  String? gameMode;
  int? ranking;
  double? distance;
  int? averagePace;
  int? calorie;
  int? duration;

  Record(
      {this.courseImgUrl,
      this.id,
      this.runStartTime,
      this.runEndTime,
      this.gameMode,
      this.ranking,
      this.distance,
      this.averagePace,
      this.calorie,
      this.duration});

  Record.fromJson(Map<String, dynamic> json) {
    if (json["courseImgUrl"] is String) {
      courseImgUrl = json["courseImgUrl"];
    }
    if (json["id"] is int) {
      id = json["id"];
    }
    if (json["runStartTime"] is String) {
      runStartTime = json["runStartTime"];
    }
    if (json["runEndTime"] is String) {
      runEndTime = json["runEndTime"];
    }
    if (json["gameMode"] is String) {
      gameMode = json["gameMode"];
    }
    if (json["ranking"] is int) {
      ranking = json["ranking"];
    }
    if (json["distance"] is double) {
      distance = json["distance"];
    }
    if (json["averagePace"] is int) {
      averagePace = json["averagePace"];
    }
    if (json["calorie"] is int) {
      calorie = json["calorie"];
    }
    if (json["duration"] is int) {
      duration = json["duration"];
    }
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> _data = <String, dynamic>{};
    _data["courseImgUrl"] = courseImgUrl;
    _data["id"] = id;
    _data["runStartTime"] = runStartTime;
    _data["runEndTime"] = runEndTime;
    _data["gameMode"] = gameMode;
    _data["ranking"] = ranking;
    _data["distance"] = distance;
    _data["averagePace"] = averagePace;
    _data["calorie"] = calorie;
    _data["duration"] = duration;
    return _data;
  }
}
